### PR TITLE
Add description to metric API responses

### LINKED ISSUE #51303: New Metrics API misses human-friendly text metric definition
### Describe the bug

https://metaboat.slack.com/archives/C0645JP1W81/p1734109505069329

### To Reproduce

https://metaboat.slack.com/archives/C0645JP1W81/p1734109505069329

### Expected behavior

_No response_

### Logs

_No response_

### Information about your Metabase installation

```JSON
51
```

### Severity

blocking some users

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lbrdnk**
Closes #51303

This PR adds `query_description` key to `:model/Card` for cards of a `:type` `:metric`. Purpose is to provide users what was known in `api/metric` responses as `definition_description`.

### CODE REVIEW

File: src/metabase/models/card.clj [hunk 1]
```diff
@@ -655,12 +655,33 @@
...
+
+  This function is used in `t2/define-after-select :model/Card`. Metadata provider caching should be considered when
+  fetching multiple metric cards having common database, as done in eg. dashboard API context."
+  [card]
+  (if-not (and (map? card)
+               (= :metric (:type card))
+               (-> card :dataset_query not-empty)
+               (-> card :database_id))
+    card
+    (or (when-some [suggested-name (some-> (lib.metadata.jvm/application-database-metadata-provider
```
Comment (appleby) [inline src/metabase/models/card.clj:671]: nit: `m/assoc-some` ?

File: src/metabase/models/card.clj [hunk 2]
```diff
@@ -655,12 +655,33 @@
       (pre-update-check-sandbox-constraints changes)
       (assert-valid-type (merge old-card-info changes)))))
 
+(defn- add-query-description-to-metric-card
+  "Add `:query_description` key to returned card.
+
+  Some users were missing description that was present in v1 metric API responses. This new key compensates for that.
+
+  This function is used in `t2/define-after-select :model/Card`. Metadata provider caching should be considered when
+  fetching multiple metric cards having common database, as done in eg. dashboard API context."
```
Comment (appleby) [inline src/metabase/models/card.clj:664]: Any particular concerns w.r.t. performance impact?

**Review by appleby — APPROVED** [review]

File: src/metabase/models/card.clj (hunk 1 above)
Comment (lbrdnk) [reply src/metabase/models/card.clj]: That makes the expression way simpler. Thank you.

File: src/metabase/models/card.clj (hunk 2 above)
Comment (lbrdnk) [reply src/metabase/models/card.clj:664]: I find it worth mentioning that if we find a context, where multiple metrics which share a database are loaded metadata provider caching could make this bit faster. That is currently relevant during dashboard load. Then, while a query is being processed. Metadata provider is also re-used there.

**Comment by lbrdnk** [comment]
@metabase-bot backport release-x.51.x

**Pull request merged**